### PR TITLE
Add interfaces for NIC, bluetooth, and audio

### PR DESCRIPTION
**Pull request by tsirakisn**
Third times a charm...

Dupe of:
https://github.com/OpenXT/vusb-daemon/pull/14
https://github.com/OpenXT/vusb-daemon/pull/29

\+ a commit for supporting USB audio

Comments addressed in separate commit; will squash when approved. See commit msg of temp commit for more info.

### CODE REVIEW

**Comment by crogers1** [comment]
I might be missing something, but where is the #include for libusb?

**Comment by tsirakisn** [comment]
> I might be missing something, but where is the #include for libusb?

@crogers1 the only method that's needed (`libusb_find_more_about_nic`) is exposed in project.h, which is already included by the necessary files.

**Comment by crogers1** [comment]
With the auto-deny now for unidentified devices, what is the user/admin process for specifying that those are in fact allowed?  Adding a new rule to the db with the corresponding device/vendor ID?

**Comment by tsirakisn** [comment]
> With the auto-deny now for unidentified devices, what is the user/admin process for specifying that those are in fact allowed? Adding a new rule to the db with the corresponding device/vendor ID?

For now, yes. Ideally we can figure out a more robust method for these devices.

File: src/libusb.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ * It parses the USB descriptors, which can have multiple
+ * configurations, which can have multiple interfaces, which can have
+ * multiple "altSettings," which can have multiple endpoints.
+ */
+void
+libusb_find_more_about_nic(device_t *device)
+{
+  int i, j, k;
+  int totalEndpoints = 0;
+  struct usb_device *libusb_device = libusb_findDevice(device->vendorid, device->deviceid);
```
Comment (jandryuk) [inline src/libusb.c]: This should have a NULL check per: https://github.com/OpenXT/vusb-daemon/pull/14#discussion_r110377798

Comment (tsirakisn) [reply src/libusb.c]: That change is in commit 1bc38df on this branch

Comment (jandryuk) [reply src/libusb.c]: Sorry, this is fixed up in commit 2.

**Review by jandryuk — APPROVED** [review]
I'm confused how `struct usb_interface_descriptor` is available.  AFAICT, libusb.h isn't included by any file - it's commented out in project.h.  configure does search for it.  Maybe it gets included transitively?

File: src/libusb.c (hunk 1 above)
Comment (jandryuk) [inline src/libusb.c]: There is trailing whitespace at the end of the libusb_findDevice(); line.  Would you please remove it?  Thanks

File: src/libusb.c [hunk 2]
```diff
@@ -121,6 +85,12 @@ libusb_find_more_about_nic(device_t *device)
   int totalEndpoints = 0;
   struct usb_device *libusb_device = libusb_findDevice(device->vendorid, device->deviceid);	
 
+  if (!libusb_device)
+  {
+    xd_log(LOG_WARNING, "Unable to find device with vendor ID %d and device ID %d. Was it removed?", device->vendorid, device->deviceid);
```
Comment (jandryuk) [inline src/libusb.c]: 4 hex digits (%04x) for VID/PID is how they are more typically printed.  I suggest you change to that.

**Review by jandryuk — CHANGES_REQUESTED** [review]
Two small fix ups I noticed while test-merging with my usbvm work.

**Comment by crogers1** [comment]
> I'm confused how `struct usb_interface_descriptor` is available. AFAICT, libusb.h isn't included by any file - it's commented out in project.h. configure does search for it. Maybe it gets included transitively?

The vusb-daemon recipe in xenclient-oe depends on ```libusb-compat``` which is a recipe defined in meta-oe.  It says it's a "drop in and replace" meant to look and feel like libusb1.  In there, ```struct usb_interface_descriptor``` is provided by ```usb.h```, which is included by ```project.h``` on line 78.

**Comment by jandryuk** [comment]
Thanks, @crogers1 -  I didn't realize usb.h was providing libusb-compat definitions.

File: src/libusb.c (hunk 1 above)
Comment (tsirakisn) [reply src/libusb.c]: Whoops, had this change locally and never pushed it. Will do.

**Comment by tsirakisn** [comment]
Addressed the two comments and squashed my temp commit into the original. Ready for merge whenever.

**Review by jandryuk — APPROVED** [review]
Thanks, @tsirakisn .  Will merge soon.

**Review by crogers1 — APPROVED** [review]

**Pull request merged**